[18.0][MIG] base_import_async: Migration to 18.0

xaviedoanhduy opened the pull request:
### Depends on:

- [18.0][MIG] queue_job: Migrate to 18.0: `https://github.com/OCA/queue/pull/692`
- [17.0][MIG] base_import_async: Migration to V17: `https://github.com/OCA/queue/pull/681`

sbidoul committed:
[IMP] rename base_import_connector to base_import_async

sbidoul committed:
[FIX] silence pylint, which btw discovered a genuine bug

sbidoul committed:
[IMP] code style using session shortcuts

sbidoul committed:
[IMP] slightly improve README

sbidoul committed:
[IMP] add comma in README

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

sbidoul committed:
[IMP] base_import_async: update README

sbidoul committed:
[IMP] OpenERP -> Odoo in comments

sbidoul committed:
[FIX] base_import_async: properly install connector module

Cfr. http://odoo-connector.com/guides/bootstrap_connector.html#install-the-module-in-the-connector

sbidoul committed:
[IMP] base_import_async: further align README with OCA template

Commit:
base_import_async: allow to set the base priority

In the options of an import. This option is too technical to be displayed on
the frontend, but the wizard can be called with the option.

It is useful when we have to run several imports one after the other and the
'other' has dependencies on the 'one'.

yvaucher committed:
Add bug tracker link on README.rst

Commit:
Replace deprecated functions

Shortcut methods session.create(), session.write(), session.browse(),
session.search() should now be directly called on session.env['model']

Commit:
Remove call to install_in_connector()

The function has been deprecated in
https://github.com/OCA/connector/commit/6e3ab38c788b003e7cdccbc949d340bd724500d7

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

lmignon committed:
[FIX] Fix new/old api compatibility
api decorator only recognize 'res_id' or 'id' as keyword args

OdooMod committed:
Update base_import_async.py

OdooMod committed:
Update base_import_async.py

sbidoul committed:
migrate base_import_async and it's test to 9.0

sbidoul committed:
migrate javascript

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

danieltorres7 committed:
Porting to new API

danieltorres7 committed:
Whitespace

lmignon committed:
[MIG] base_import_async: replace connector by queue_job

lmignon committed:
[FIX] base_import_async: with_delay returns an instance of Job

sbidoul committed:
[MIG] queue_job_subscribe 10.0

sbidoul committed:
[IMP] queue_job_subscribe: style

sbidoul committed:
[IMP] queue_job_subscribe: odoo tag in views

sbidoul committed:
[IMP] queue_job_subscribe: improve README

pedrobaeza committed:
[MIG] Make modules uninstallable

Commit:
Migrate queue_job_subscribe to v11

Commit:
[ADD] add new module to execute ir_cron as queue_job

Commit:
Improve tests coverage

sbidoul committed:
queue_job_cron: do not require read access to ir.cron

sbidoul committed:
queue_job_cron: bump version

Commit:
Correct or disable pylint warnings

tarteo committed:
[MIG] base_import_async and test_base_import_async to 11.0

tarteo committed:
[IMP] Processed review

tarteo committed:
[IMP] Readability, speed, and added HISTORY.rst

OCA-git-bot committed:
[MIG] Make modules uninstallable

OCA-git-bot committed:
[MIG] Make modules uninstallable

Commit:
Migrate base_import_async to 12.0

Commit:
Add OCA development status

ThomasBinsfeld committed:
[MIG] Queue Job Cron to 12.0

oca-travis committed:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[ADD] icon.png

Commit:
Store dependencies

Commit:
Add wait dependencies state

marylla committed:
Added translation using Weblate (German)

Commit:
Add API for Delayables

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (4 of 4 strings)

Translation: queue-12.0/queue-12.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-12-0/queue-12-0-queue_job_cron/de/

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (4 of 4 strings)

Translation: queue-12.0/queue-12.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-12-0/queue-12-0-queue_job_cron/zh_Hans/

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Rename Chinese translation files

Commit:
Migrate queue_job_cron to 13.0

Commit:
Migrate queue_job_subscribe to 13.0

Commit:
Set modules as uninstallable

Commit:
Set modules as uninstallable

Commit:
Run pre-commit with black, isort, ...

Commit:
Run pre-commit with black, isort, ...

Commit:
Run pre-commit with black, isort, ...

oca-travis committed:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
Exit the import screen after importing file in async

When calling super, the super method would fail with a 'state machine'
error, appearing confusingly as a CORS error client side. We do not
have to call super when we delayed import asynchronously, as anyway
the 'super' call would send an inaccurate confirmation notification and exit.

Commit:
Set default values for encoding and separator

Import of Excel files do not set these options, although the keys
are in the dict.

Courtesy of @GSLabIt for finding the issue and solution.

OCA-git-bot committed:
[ADD] icon.png

Commit:
[IMP] base_import_async: black, isort

Commit:
[MIG] base_import_async: Migration to 13.0

Commit:
Apply suggestions from code review

Co-Authored-By: Sébastien Alix <sebalix@users.noreply.github.com>

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

Commit:
[FIX] base_import_async error message

OCA-git-bot committed:
base_import_async 13.0.1.0.1

Commit:
Add method to patch a method to be automatically delayed

This patch method has to be called in ``_register_hook``.

When a method is patched, any call to the method will not directly
execute the method's body, but will instead enqueue a job.

When a ``context_key`` is set when calling ``_patch_job_auto_delay``,
the patched method is automatically delayed only when this key is
``True`` in the caller's context. It is advised to patch the method
with a ``context_key``, because making the automatic delay *in any
case* can produce nasty and unexpected side effects (e.g. another
module calls the method and expects it to be computed before doing
something else, expecting a result, ...).

A typical use case is when a method in a module we don't control is called
synchronously in the middle of another method, and we'd like all the calls
to this method become asynchronous.

It relies on https://github.com/OCA/queue/pull/274 that deprecates the
`@job` decorator.

Commit:
queue_job_cron: remove `@job` decorator

Commit:
base_import_async: remove `@job` decorator

Commit:
Fix invalid development_status (accepted value is Production/Stable)

Commit:
Fix channel name in jobs delayed for crons

The channel must be 'root.ir_cron', not 'ir_cron'.

OCA-git-bot committed:
base_import_async 13.0.2.0.0

OCA-git-bot committed:
queue_job_cron 13.0.2.0.0

OCA-git-bot committed:
queue_job_cron 13.0.2.1.0

sebastienbeau committed:
[MIG] init V14 repo

- set module uninstallable
- add new .dot file config for v14

sebastienbeau committed:
[MIG] init V14 repo

- set module uninstallable
- add new .dot file config for v14

sebastienbeau committed:
[MIG] init V14 repo

- set module uninstallable
- add new .dot file config for v14

sebastienbeau committed:
[IMP] : black, isort, prettier

sebastienbeau committed:
[IMP] : black, isort, prettier

oca-travis committed:
[UPD] Update test_queue_job.pot

Commit:
Remove initial create notification and follower

Everytime a job is created, a mail.message "Queue Job created" is
created. This is useless, as we already have the creation date and user,
and nobody will ever want to receive a notification for a created job
anyway.

Removing the on creation auto-subscription of the user that created the
job makes sense as well since we automatically subscribe the queue job
managers for failures. Add the owner of the jobs to the followers on
failures only as well.

It allows to remove a lot of insertions of records (and of deletions
when autovacuuming jobs).

simahawk committed:
queue_job: add exec time to view some stats

codingforfun committed:
[MIG] Bump version. Set module installable. Check migration guide.

codingforfun committed:
[FIX] Respect `cron.user_id` for manually triggered queue jobs

PieterPaulussen committed:
[REF] Replace deprecated assertEquals with assertEqual

PieterPaulussen committed:
[IMP] Store job execution information + wizard to terminate a job + test

simahawk committed:
queue_job: store exception name and message

simahawk committed:
queue_job: add hook to customize stored values

PieterPaulussen committed:
[FIX] use __func__ for function comparison in tests

Commit:
[ADD] Apply suggestions from code review

Co-authored-by: Simone Orsi <simahawk@users.noreply.github.com>

oca-travis committed:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update test_queue_job.pot

OCA-git-bot committed:
test_queue_job 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-14.0/queue-14.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-14-0/queue-14-0-test_queue_job/

guewen committed:
Revert "[14.0] [IMP] Store job execution information + wizard to terminate a job + test"

oca-travis committed:
[UPD] Update test_queue_job.pot

OCA-git-bot committed:
test_queue_job 14.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-14.0/queue-14.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-14-0/queue-14-0-test_queue_job/

guewen committed:
Add a graph UUID

A graph of jobs always share the same graph_uuid, which can be
used to group jobs, but is also a faster way to find all the jobs
of the graph. Then we can use the dependencies field to build the
whole graph from the pre-selection of jobs.

guewen committed:
Ignore requeues on dependency jobs waiting on parent jobs

Jobs waiting that their dependencies are executed cannot be requeued.
They have to keep waiting their turn.

OCA-git-bot committed:
test_queue_job 14.0.1.3.0

BT-jcamacho committed:
[MIG] base_import_async: Migration to 14.0

guewen committed:
Fix warnings in tests

As jobs are created in the setup, but not always delayed,
the __del__ warning that the job was not delayed is triggered.
Create only the jobs necessary for each test.

guewen committed:
Improve display of jobs graph widget

guewen committed:
Add powerful context manager for running tests on jobs

* mock_jobs context manager with assert methods to check enqueued jobs
* jobs are not stored in database, yet they can be inspected and performed
  during the test
* the context manager works with graphs enqueued by delayable and with with_delay()
* direct execution of jobs works with graphs by executing them synchronously
  following their topological sort

guewen committed:
Set graph_uuid only once in DelayableGraph

The previous implementation was incomplete, as it was setting
the graph_uuid when `Job.add_depends()` was called. It had the
advantage of being correct when (for e.g. a test) manually created
a Job and called `add_depends()`, but it could not work when in
some situations involving a group where one of the job has no job
depending on it.

Also, remove ``Job.add_reverse_depends()`` which is not used.

guewen committed:
Add docstrings on the new delayable classes

LoisRForgeFlow committed:
[13.0][FIX] queue_job_cron: channel_id must be storable.

Otherwise, you cannot use any channel other than default (
root.ir_cron)

sbejaoui committed:
[14.0][MIG] - queue_job_subscribe

acsonefho committed:
[14.0][IMP] queue_job current company
Use the current company to trigger the job (+ add related tests)

[14.0][FIX] queu_job: allowed_company_ids => use with_company(...)
Fill allowed_company_ids from context with the job's company instead of every allowed companies of the user.
Because most of the time, a job is related to only one company. And adding every allowed companies of the user into the context may load some unexpected records (during search for example). Because standards ir.rule use ['|',('company_id','=',False),('company_id', 'in', company_ids)] and this 'company_ids' is filled with every allowed companies from the context.

OCA-git-bot committed:
base_import_async 14.0.1.0.1

guewen committed:
Set modules as uninstallable

guewen committed:
Set modules as uninstallable

guewen committed:
Set modules as uninstallable

guewen committed:
Set modules queue_job and test_queue_job as installable

guewen committed:
Run pre-commit run -a

guewen committed:
[MIG] test_queue_job: Migration to 15.0

Commit:
[UPD] Update test_queue_job.pot

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 8.3% (1 of 12 strings)

Translation: queue-14.0/queue-14.0-base_import_async
Translate-URL: https://translation.odoo-community.org/projects/queue-14-0/queue-14-0-base_import_async/it/

guewen committed:
Rename mock_jobs() to trap_jobs()

The fact that mocks are used is an implementation detail, and is actually
hidden to the end user of the test API.

guewen committed:
Rename dependency method done() to on_done()

on_done() is maybe a better choice, it is less ambiguous than then() (clearer
on the fact it happens on done()) and done() (less confusion with set_done()).
I'll do the renaming soon, unless you have other suggestions.

Ref: https://github.com/OCA/queue/pull/154#issuecomment-1025127060

JoanMForgeFlow committed:
[MIG][15.0] queue_job_cron to version 15.0

JoanMForgeFlow committed:
[MIG][15.0] queue_job_cron to version 15.0

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

Commit:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[UPD] README.rst

guewen committed:
Disable storing of context in jobs recordsets by default

It can still be activated by overriding a method.

SilvioC2C committed:
[MIG] queue_job_subscribe: Migration to 15.0

Commit:
base_import_async: add a dedicated channel

baimont committed:
[MIG] queue_job_cron: Migration to 16.0

sbidoul committed:
Initialize 16.0

sbidoul committed:
Initialize 16.0

baimont committed:
[MIG] test_queue_job: Migration to 16.0

baimont committed:
[FIX] import mock from unittest

Commit:
[UPD] Update test_queue_job.pot

Commit:
[UPD] Update test_queue_job.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-16.0/queue-16.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-test_queue_job/

sbidoul committed:
Update test with proper way to pass company to job

guewen committed:
Apply pre-commit on migration of jobs graph

guewen committed:
Use new TransactionCase instead of deprecated SavepointCase

guewen committed:
Fix things required by odoo 14.0 or python 3.9

* Renaming the TestXxx models in test_queue_job is because of pytest that
  complains, confusing them for test classes
* logger.warn is deprecated in favor of logger.warning
* Add invalidation of cache in SQL update that enqueue dependent jobs
* Flush in a test, the same is done currently in the controller
* Weird one: we can no longer compare the bound methods of the models,
  compare their __func__ which is equal.

OCA-git-bot committed:
test_queue_job 16.0.1.0.1

Commit:
[UPD] Update test_queue_job.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-15.0/queue-15.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-15-0/queue-15-0-test_queue_job/

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 75.0% (9 of 12 strings)

Translation: queue-16.0/queue-16.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-test_queue_job/de/

lmignon committed:
[MIG] queue_job, test_queue_job: Forward port queue from 15.0

This forward port takes all the changes from 15.0 not yet into 16.0
up to https://github.com/OCA/queue/commit/173b6ca7026dc8042cc8e750767c3267c67f4f43

This FP contains the new queue dependencies functionality.
The dedicated widget used to display the dependecies graph on the job
form is also migrated to owl

OCA-git-bot committed:
test_queue_job 16.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-16.0/queue-16.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-test_queue_job/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
base_import_async 14.0.1.0.2

dhongu committed:
Added translation using Weblate (Romanian)

ieski committed:
[IMP] base_import_async: black, isort, prettier

ieski committed:
[MIG] base_import_async: Migration to 15.0

simahawk committed:
queue_job: unify no delay option

Current situation:

* multiple keys for no good reason
* half baked: not all of them used everywhere
* no centralization
* poor naming

With this change we'll have:

* 1 and only one key to disable via ctx: ``queue_job__no_delay``
* 1 and only one key to disable via os env: ``QUEUE_JOB__NO_DELAY``
* backward compatibility with deprecation for old keys

dzungtran89 committed:
[16.0][MIG] base_import_async: Migration to 16.0

OCA-git-bot committed:
test_queue_job 16.0.2.1.0

marielejeune committed:
[FIX] base_import_async: remove default values from context when creating the attachment.

Some default values might be present in context depending on the action we came from
when clicking on 'import' button. These default values are not intended to
be default values for the ir.attachment record.
In some cases they cause an error because a field with the same name exists on
ir.attachment, as for e.g. the 'default_type'='opportunity' value
present in the standard crm.lead action context.

guewen committed:
Fix trap_jobs() recordset comparison

The trap_jobs() test helper accepts any model/recordset as long
as the job function is the same, which is not enough to verify that
the job has been delayed on the expected recordset or model.

Tested job functions are "bound methods" so we can compare their `__self__`.

Example of assertion that would be valid:

  partner = self.env["res.partner"].create({"name": "foo"})
  with trap_jobs() as trap:
    partner.with_delay().do_something(42)
    trap.assert_enqueued_job(
        self.env["res.partner"].do_something,
        args=(42,)
    )
    trap.assert_jobs_count(1, only=self.env["res.partner"].do_something)

Now, it would fail with an error message like:

E           AssertionError: Job <res.partner()>.do_something(42) with properties (channel=root.test, description=Do something, eta=15, identity_key=<function identity_exact at 0x7f01bfb54cb0>, max_retries=1, priority=15) was not enqueued.
E           Actual enqueued jobs:
E            * <res.partner(64,)>.do_something(42) with properties (priority=15, max_retries=1, eta=15, description=Do something, channel=root.test, identity_key=<function identity_exact at 0x7f01bfb54cb0>)

ramiadavid committed:
[MIG] queue_job_subscribe: Migration to 16.0

OCA-git-bot committed:
test_queue_job 16.0.2.2.0

Commit:
[IMP] queue_job: identity_key enhancements

1. In production, a job which is waiting dependencies or which
has started, but not completed, should not be repeated if
the identity_key matches.

2. In tests, the mock queue handler is now enhanced to allow
better mimicking of the identity_key blocks from production.

3. In tests, the mock queue handler now clears the enqueued jobs
after performing them, to better reproduce what a production
environment would do.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-16.0/queue-16.0-base_import_async
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-base_import_async/

kikopeiro committed:
Added translation using Weblate (Spanish)

kikopeiro committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: queue-16.0/queue-16.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-queue_job_cron/es/

OCA-git-bot committed:
[UPD] README.rst

danielduqma committed:
[FIX] base_import_async: restore class

In migration to 16.0, input class `oe_import_queue` was replaced by
another one when it has to be added instead, causing the module not
to work. In Javascript, input is searched by this class and it's not
found, so value cannot be recovered and import is made
synchronously.

OCA-git-bot committed:
oca-github-bot post-merge updates

asierneiradev committed:
Added translation using Weblate (Spanish)

asierneiradev committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (9 of 9 strings)

Translation: queue-16.0/queue-16.0-base_import_async
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-base_import_async/es/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-16.0/queue-16.0-base_import_async
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-base_import_async/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-16.0/queue-16.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-queue_job_cron/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (13 of 13 strings)

Translation: queue-16.0/queue-16.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-test_queue_job/es/

OCA-git-bot committed:
[BOT] post-merge updates

nguyenminhchien committed:
[IMP] test_queue_job: pre-commit auto fixes

nguyenminhchien committed:
[MIG] test_queue_job: Migration to 17.0

Commit:
[UPD] Update test_queue_job.pot

ramiadavid committed:
[IMP] queue_job_subscribe: pre-commit auto fixes

ramiadavid committed:
[MIG] queue_job_subscribe: Migration to 17.0

lmignon committed:
[IMP] queue_job_cron: Avoid parallel run

By default, odoo never runs the same cron job in parallel. This commit uses the identity key mechanism to enforce this mechanism when a cron job is run as a queue job. This behaviour can be controlled by a new setting on the cron definition but is activated by default to keep the original behaviour

Commit:
[UPD] Update queue_job_subscribe.pot

mymage committed:
Added translation using Weblate (Italian)

nguyenminhchien committed:
[IMP] queue_job_cron: Pre-commit auto fixes

nguyenminhchien committed:
[MIG] queue_job_cron: Migration to 17.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: queue-16.0/queue-16.0-base_import_async
Translate-URL: https://translation.odoo-community.org/projects/queue-16-0/queue-16-0-base_import_async/it/

Commit:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[BOT] post-merge updates

alexbottemanne committed:
[16.0][FIX] base_import_async: update OPT_HAS_HEADERS option

InigoGR committed:
queue_job: fix retry format with tuple values

#348 added support for randomized retry intervals.
Configuration of randomized retry intervals is not possible due to the formatting checks not being updated.
This should fix it.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-17.0/queue-17.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-test_queue_job/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: queue-17.0/queue-17.0-queue_job_subscribe
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-queue_job_subscribe/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: queue-17.0/queue-17.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-queue_job_cron/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: queue-17.0/queue-17.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-test_queue_job/it/

pedrobaeza committed:
[FIX] *queue_job: Pre-commit fixes

OCA-git-bot committed:
[BOT] post-merge updates

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (5 of 5 strings)

Translation: queue-17.0/queue-17.0-queue_job_subscribe
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-queue_job_subscribe/zh_CN/

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (12 of 12 strings)

Translation: queue-17.0/queue-17.0-test_queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-test_queue_job/zh_CN/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: queue-17.0/queue-17.0-queue_job_subscribe
Translate-URL: https://translation.odoo-community.org/projects/queue-17-0/queue-17-0-queue_job_subscribe/pt_BR/

ChrisOForgeFlow committed:
[16.0][FIX] base_import_async, add sudo call to models without explicit access

QuocDuong1306 committed:
[FIX] queue_job: missing flush

- an explicit flush is needed or child jobs won't be updated
- no need to forward port, this was fixed already in 16.0+

QuocDuong1306 committed:
[IMP] queue_job: Cancel child jobs when the parent is cancelled

rousseldenis committed:
[IMP] queue_job: Don't raise a warning for valid context

As the 'queue_job__no_delay' context key is the valid one to use,
don't raise warnings during tests, but log it as information level.

Wodran14 committed:
[IMP] base_import_async: pre-commit stuff

Wodran14 committed:
[MIG] base_import_async: Migration to 17.0

QuocDuong1306 committed:
[MIG] test_queue_job: Migration to 18.0

QuocDuong1306 committed:
[MIG] queue_job_cron: Migration to 18.0

QuocDuong1306 committed:
[MIG] queue_job_subscribe: Migration to 18.0

xaviedoanhduy committed:
[IMP] base_import_async: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] base_import_async: Migration to 18.0

chaule97 commented:
please write one testcase

xaviedoanhduy commented on base_import_async/static/src/xml/import_data_sidepanel.xml:
```diff
@@ -2,11 +2,12 @@
 <templates xml:space="preserve">
     <t
         t-name="base_import_async.ImportDataSidepanel"
-        t-inherit="base_import.ImportDataSidepanel"
```
I found that the following template inheritance syntax seems to be correct according to the documentation, but when installing the module it shows an error in the Console interface "`Missing (extension) parent templates odoo: base_import.ImportDataSidepanel`", so when I remove the `base_import.` prefix, the module works

thienvh332 committed:
[UPD] queue_job_subscribe: Update disable tracking and use no_reset_password for tests

xaviedoanhduy committed:
[ADD] base_import_async: add tests

xaviedoanhduy commented:
Hello @chaule97, I have added test

chaule97 reviewed (approved):
Pre-approve:

Can you set [[DON'T MERGE] Add test-requirements.txt](https://github.com/xaviedoanhduy/queue/pull/3/commits/408a6c62efca9e11c9105100b532abf615cbdd4e) as the last commit

OCA-git-bot committed:
Merge PR #692 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update queue_job.pot

Commit:
[UPD] Update test_queue_job.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-18.0/queue-18.0-queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-18-0/queue-18-0-queue_job/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (156 of 156 strings)

Translation: queue-18.0/queue-18.0-queue_job
Translate-URL: https://translation.odoo-community.org/projects/queue-18-0/queue-18-0-queue_job/it/

OCA-git-bot committed:
Merge PR #693 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update queue_job_cron.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-18.0/queue-18.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-18-0/queue-18-0-queue_job_cron/

OCA-git-bot committed:
Merge PR #695 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update queue_job_subscribe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: queue-18.0/queue-18.0-queue_job_subscribe
Translate-URL: https://translation.odoo-community.org/projects/queue-18-0/queue-18-0-queue_job_subscribe/

OCA-git-bot committed:
[IMP] update dotfiles

thienvh332 committed:
[FIX] queue_job: fix test cases from TestDelayable being skipped

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: queue-18.0/queue-18.0-queue_job_cron
Translate-URL: https://translation.odoo-community.org/projects/queue-18-0/queue-18-0-queue_job_cron/it/

guewen committed:
Fix SQL injection lint in 18.0.1.0.0 pre-migrate.py

guewen committed:
Fix 'document' is not defined [no-undef] lint in queue_job

guewen committed:
Merge pull request #701 from OCA/18.0-ocabot-update-dotfiles

[18.0] dotfiles update needs manual intervention

OCA-git-bot committed:
Merge PR #702 into 18.0

Signed-off-by guewen

xaviedoanhduy committed:
Fix 'document' is not defined [no-undef] lint in base_import_async
